src/interface/mod.rs

```diff
@@ -278,64 +304,54 @@ pub(crate) enum InterfaceType {
 
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub(crate) struct DatastreamIndividual {
-    mappings: MappingMap<'static, DatastreamIndividualMapping>,
+    mappings: MappingVec<DatastreamIndividualMapping>,
 }
 
 impl DatastreamIndividual {
-    pub fn iter_mappings(&self) -> IndividualMappingIter {
+    pub(crate) fn iter_mappings(&self) -> IndividualMappingIter {
         IndividualMappingIter::new(&self.mappings)
     }
 
-    pub fn add_mapping(&mut self, mapping: &Mapping) -> Result<(), InterfaceError> {
-        let individual = DatastreamIndividualMapping::try_from(mapping)?;
+    pub(crate) fn add_mapping(
+        tree: &mut MappingSet<DatastreamIndividualMapping>,
```

Comment:
Why didn't use `&self` on this method? So you can put `tree` as internal method, without pass it as external argument.

Reply:
The tree is not internal to the struct, there are two types of structures MappingSet and MappingVec (the names are unfortunately similar), the tree is needed during contruction of the interface to ensure that the mappings are unique, while the vec is used to retrieve a mapping comparing the endpoints. 

This is a static method on Self since is dependent on the type of interface and is used to add a mapping to the three, when all mapping are added the MappingSet will then be converted into a MappingVec